shell/platform/android/io/flutter/view/FlutterNativeView.java

```diff
@@ -193,52 +197,47 @@ private void handlePlatformMessageResponse(int replyId, byte[] reply) {
 
     // Called by native to update the semantics/accessibility tree.
     private void updateSemantics(ByteBuffer buffer, String[] strings) {
-        if (mFlutterView == null)
-            return;
+        if (mFlutterView == null) return;
         mFlutterView.updateSemantics(buffer, strings);
     }
 
     // Called by native to notify first Flutter frame rendered.
     private void onFirstFrame() {
-        if (mFlutterView == null)
-            return;
+        if (mFlutterView == null) return;
         mFlutterView.onFirstFrame();
     }
 
-    private static native long nativeAttach(FlutterNativeView view);
+    private void onStarted(boolean success) {
+        if (startedEvent == null) return;
+        startedEvent.onStarted(success);
```

Comment:
What is this event used for?  Can this be replaced by a message sent by the app on a platform channel at a time of the app's choosing?

This will be called after the engine invokes _startMainIsolate, and the app code may not be fully initialized when this callback happens.

Reply:
This event is used to notify that the either the new shell has started running or there was an error starting the shell. Other than having this event, I don't believe that the plugin can determine whether or not the new shell was started successfully.

Good point about this being called after _startMainIsolate is invoked when the app still may not be fully initialized, but we could totally use a platform channel to notify the plugin when the background callback determines that initialization is complete.